### PR TITLE
Installer updates for JDK20 for July 2023 release

### PR DESCRIPTION
**Pull request by adamfarley**
Styled after the [April PR](https://github.com/adoptium/installer/pull/662/files)

### CODE REVIEW

**Review by steelhead31 — APPROVED** [review]
LGTM

**Comment by jiekang** [comment]
Debian failing relevant log:
```
# Download and verify checksum of JRE
wget --progress=dot:mega -O jre.tar.gz https://github.com/adoptium/temurin20-binaries/releases/download/jdk-20.0.1%2B9/OpenJDK20U-jre_x64_linux_hotspot_20.0.2_9.tar.gz
Resolving github.com (github.com)... 140.82.113.4
Connecting to github.com (github.com)|140.82.113.4|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
make[1]: Leaving directory '/home/builder/workspace/packaging'
2023-07-26 12:11:12 ERROR 404: Not Found.
```

File: linux/jre/debian/src/main/packaging/temurin/20/debian/rules [hunk 1]
```diff
@@ -3,10 +3,10 @@
 pkg_name = temurin-20-jre
 priority = 1912
 jvm_tools = java jfr jrunscript jwebserver keytool rmiregistry
-amd64_tarball_url = https://github.com/adoptium/temurin20-binaries/releases/download/jdk-20.0.1%2B9/OpenJDK20U-jre_x64_linux_hotspot_20.0.1_9.tar.gz
-amd64_checksum = daacf24c15bf7f38a957a98a312911a36ba7f7d97004920a7875791f20e8e1ed
-arm64_tarball_url = https://github.com/adoptium/temurin20-binaries/releases/download/jdk-20.0.1%2B9/OpenJDK20U-jre_aarch64_linux_hotspot_20.0.1_9.tar.gz
-arm64_checksum = 4b04fcfabf833403cc74dd19105a387563f9ff0fef975c4101f3d74c53eb7745
+amd64_tarball_url = https://github.com/adoptium/temurin20-binaries/releases/download/jdk-20.0.1%2B9/OpenJDK20U-jre_x64_linux_hotspot_20.0.2_9.tar.gz
```
Comment (jiekang) [inline linux/jre/debian/src/main/packaging/temurin/20/debian/rules:6]: s|download/jdk-20.0.1|download/jdk-20.-.2| ...

**Review by jiekang — CHANGES_REQUESTED** [review]

File: linux/jre/debian/src/main/packaging/temurin/20/debian/rules (hunk 1 above)
Comment (adamfarley) [reply linux/jre/debian/src/main/packaging/temurin/20/debian/rules]: I was just fixing that. :)
Good catch Jie.

**Review by gdams — APPROVED** [review]

**Pull request merged**